core/benches/bls_sigverify.rs

```diff
@@ -20,18 +20,54 @@ use {
         consensus_message::{Certificate, ConsensusMessage, VoteMessage},
         vote::Vote,
     },
-    std::{cell::RefCell, sync::Arc},
+    std::{
+        cell::RefCell,
+        sync::{
+            atomic::{AtomicUsize, Ordering},
+            Arc,
+        },
+    },
 };
 
 const BENCH_SLOT: u64 = 70;
 // TODO(sam): use a small number for now to emulate the current test cluster
 const NUM_VALIDATORS: usize = 50;
 
+const NUM_PREGENERATED_BATCHES: usize = 500;
```

Comment:
- Since the root slot is not advancing during the benchmark, these will all get cached, right?  So we are not going to be benchmarking the cost of serializing the vote messages.  I wonder if we should include this information somewhere in this file.  
- Also maybe add some intuition for how the number 500 was picked.
- finally, should we use explicit warm up in the benchmarking below to ensure that all 500 batches get cached before we start the actual benchmarking?

Reply:
These are good comments. I added a note stating that the benchmarks excludes the overhead of serialization. Maybe we can add benches to measure serialization times in a follow-up.

I just chose 500 since it is a large enough to offer sufficient unique inputs to saturate the caches, but small enough to keep the memory overhead of the pool reasonable. I added a comment regarding this, but we can always tweak this number later on.

Regarding explicit warm ups, I think criterion should actually take care of this. The default warm-up phase (3 seconds) should be sufficient to cycle through all 500 unique inputs multiple times to ensure that the caches are hot before measurement begins.